src/AggregateFunctions/AggregateFunctionSparkbar.h

```diff
@@ -105,24 +107,12 @@ class AggregateFunctionSparkbar final
     bool specified_min_max_x;
 
     template <class T>
-    String getBar(const T value) const
+    int updateFrame(ColumnString::Chars & frame, const T value) const
```

Comment:
`updateFrame` is more complicated than `getBar`, what is the key difference?

Reply:
Before we get a string (which create a copy), insert it into the the column (which is another copy). `updateFrame` will keep a static list of strings, then select the proper value from the list, then insert it into the column.